File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/SKActionRun.java
package br.com.insanitech.spritekit;

/**
 * Created by anderson on 06/01/17.
 */

class SKActionRun extends SKAction {
    private Runnable runnable;

    public SKActionRun(Runnable runnable) {
        this.runnable = runnable;
    }

    @Override
    void computeStart() {

    }

    @Override
    void computeAction(long elapsed) {

    }

    @Override
    void computeFinish() {
        if (runnable != null) {
            runnable.run();
        }
    }

    @Override
    boolean willHandleFinish() {
        return false;
    }
}

File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/SKActionRun.kt
package br.com.insanitech.spritekit

/**
 * Created by anderson on 06/01/17.
 */
internal class SKActionRun(private val runnable: SKBlock?) : SKAction() {

    internal override fun computeStart() {

    }

    internal override fun computeAction(elapsed: Long) {

    }

    internal override fun computeFinish() {
        runnable?.invoke()
    }
}
